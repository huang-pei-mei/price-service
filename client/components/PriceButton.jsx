import React from 'react';

const PriceButton = (props) => (
  <div id='price'>
    <button className='priceButton' id='bookBuy'>Buy for ${props.price ? props.price.toFixed(2) : ''}</button>
  </div>
);

export default PriceButton;