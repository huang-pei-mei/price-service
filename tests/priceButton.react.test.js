import React from 'react';
import { shallow, render } from 'enzyme';
import PriceButton from '../client/components/PriceButton.jsx';

describe('PriceButton', () => {
  it('should render component', () => {
    const component = shallow(<PriceButton />);

    expect(component.getElements()).toMatchSnapshot();
  });

  it('should render price when passed in props', () => {
    const priceButtonProps = {
      price: 36.00
    };
    const component = shallow(<PriceButton {...priceButtonProps}/>);

    expect(component.text()).toContain([priceButtonProps.price].toString());
  });
});