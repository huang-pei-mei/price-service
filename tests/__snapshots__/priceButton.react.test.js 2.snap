// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceButton should render component 1`] = `
Array [
  <div
    id="price"
  >
    <button
      className="priceButton"
      id="bookBuy"
    >
      Buy for $
      
    </button>
  </div>,
]
`;
